lib/core/elastic_connector_actions.rb

```diff
@@ -88,6 +88,38 @@ def claim_job(connector_id)
         job['_id']
       end
 
+      def find_current_job(connector_id)
+        response = client.search(
+          :index => Utility::Constants::JOB_INDEX,
+          :ignore => 404,
+          :body => {
+            :size => 1,
+            :from => 0,
+            :query => {
+              :bool => {
+                :must => [
+                  {
+                    :term => {
+                      :connector_id => connector_id
+                    }
+                  },
+                  {
+                    :term => {
+                      :status => Connectors::SyncStatus::IN_PROGRESS
+                    }
+                  }
+                ]
+              }
+            },
+            :sort => [
+              'created_at' => { :order => 'desc' }
+            ]
+          }
+        )
+        hits = response['hits']['hits']
+        hits.empty? ? nil : hits[0].with_indifferent_access
```

Comment:
do we need an `ensure` block to protect against any non-200 responses? What about some null checks in case `[hits][hits]` isn't the structure of the response?

Reply:
I've added the null checks now. As to the non-200, I am ignoring the 404 right now (treat it like an empty set). I would assume that other error codes should fail the whole method. Which ones do you think we need to catch?

Reply:
I'm totally fine with them failing the method. I'm mostly worried that if you get a:
```
{
  "errors": ["this is a very important error message"]
}
```

response that we'll instead throw an NPE (you've fixed that) or just give an empty result, instead of actually acknowledging that the response indicated an error.

Typically I wouldn't nitpick like this, but since it's so late in the dev cycle, I think we should either wait on this PR for another release or have a REALLY high bar for ensuring it doesn't add any risk.

Reply:
@seanstory I suspect that the [ship might have sailed on 8.5 anyway](https://github.com/elastic/enterprise-search-team/issues/2913#issuecomment-1266889446), but I will try to make it more bullet-proof in case it has not. =)